metrics-facade-prometheus/src/main/java/com/ringcentral/platform/metrics/samples/prometheus/collectorRegistry/SimpleCollectorRegistryPrometheusInstanceSamplesProvider.java

```diff
@@ -0,0 +1,143 @@
+     * @param collectorRegistries must be not empty
+     */
+    public SimpleCollectorRegistryPrometheusInstanceSamplesProvider(
+        MetricName namePrefix,
+        Predicate<String> metricFamilyNameFilter,
+        Predicate<String> metricFamilySampleNameFilter,
+        CollectorRegistry... collectorRegistries) {
+
+        this(namePrefix, metricFamilyNameFilter, metricFamilySampleNameFilter, List.of(collectorRegistries));
+    }
+
+    /**
+     * @param namePrefix the prefix to be prepended to each MetricFamilySample.name when building PrometheusInstanceSample.instanceName/name
+     *                   and to be prepended to each MetricFamilySample.Sample.name when building PrometheusSample.name.
+     *                   Optional
+     *
+     * @param metricFamilyNameFilter optional
+     * @param metricFamilySampleNameFilter optional
+     * @param collectorRegistries must be not empty
+     */
+    public SimpleCollectorRegistryPrometheusInstanceSamplesProvider(
+        MetricName namePrefix,
+        Predicate<String> metricFamilyNameFilter,
+        Predicate<String> metricFamilySampleNameFilter,
+        Collection<? extends CollectorRegistry> collectorRegistries) {
+
+        this.namePrefix = namePrefix;
+        this.metricFamilyNameFilter = metricFamilyNameFilter;
+        this.metricFamilySampleNameFilter = metricFamilySampleNameFilter;
+
+        checkArgument(
+            collectorRegistries != null && !collectorRegistries.isEmpty(),
+            "collectorRegistries is null or empty");
+
+        this.collectorRegistries = collectorRegistries;
+    }
+
+    @Override
+    public Set<PrometheusInstanceSample> instanceSamples() {
+        Set<PrometheusInstanceSample> result = new LinkedHashSet<>();
+        collectorRegistries.forEach(collectorRegistry -> processCollectorRegistry(collectorRegistry, result));
+        return result;
+    }
+
+    private void processCollectorRegistry(CollectorRegistry collectorRegistry, Set<PrometheusInstanceSample> result) {
+        Enumeration<MetricFamilySamples> fsEnumeration =
+            metricFamilyNameFilter != null ?
+            collectorRegistry.filteredMetricFamilySamples(metricFamilyNameFilter) :
+            collectorRegistry.metricFamilySamples();
+
+        while (fsEnumeration.hasMoreElements()) {
+            MetricFamilySamples fs = fsEnumeration.nextElement();
+
+            MetricName name =
+                hasNamePrefix() ?
+                MetricName.of(namePrefix, fs.name) :
+                MetricName.of(fs.name);
+
+            PrometheusInstanceSample instanceSample = new PrometheusInstanceSample(name, name, fs.help, fs.type);
+
+            fs.samples.forEach(fsSample -> {
+                if (metricFamilySampleNameFilter != null && !metricFamilySampleNameFilter.test(fsSample.name)) {
+                    return;
+                }
+
+                MetricName sampleName =
+                    hasNamePrefix() ?
```

Comment:
I think it could be checked once in constructor and remembered in boolean hasNamePrefix field.

Or MetricNameBuilder function could be initiated in constructor and used to create MetricName instances afterwards, then neither ternary operator nor hasNamePrefix won't be needed. Actually in this case namePrefix field could be removed too as the function will have it.

Reply:
Agree, will add a name builder.

Reply:
Added the builder.